hack/release/main.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"regexp"
+
+	"sigs.k8s.io/kubetest2/pkg/process"
+)
+
+var (
+	dir          string
+	version      string
+	versionRegxp = regexp.MustCompile(`^\d+\.\d+\.\d+(-beta\.\d+)?$`)
+)
+
+func main() {
+	flag.StringVar(&dir, "dir", ".", "the directory run in")
+	flag.StringVar(&version, "version", "", "the new version to release")
+	flag.Parse()
+
+	steps := []Step{
+		ValidateVersion(),
+		UpdateVersion(),
+		CreateReleaseBranch(process.ExecJUnit),
+		GenerateFiles(process.ExecJUnit),
```

Comment:
Maybe also ad a step to make sure we change the version? In other words, that the version in `version.txt` is not the same passed. I have vague memories that the `make release/gen-files` fails if you rerun it...

Reply:
It does indeed. Nice catch. I'll add a check

Reply:
I added a check to verify the new version against the existing git tags